schema/canonical/crc.go

```diff
@@ -46,12 +27,53 @@ func init() {
 	}
 }
 
-func AvroCRC64Fingerprint(data []byte) string {
+func AvroCRC64Fingerprint(schema []byte, writer io.Writer) error {
 	fp := EMPTY
-	for _, d := range data {
+	for _, d := range schema {
 		fp = (fp >> 8) ^ FP_TABLE[(fp^uint64(d))&0xff]
 	}
-	output := bytes.NewBuffer(make([]byte, 0, 8))
-	binary.Write(output, binary.LittleEndian, fp)
-	return hex.EncodeToString(output.Bytes())
+	err := binary.Write(writer, binary.LittleEndian, fp)
+	return err
+}
+
+func AvroCalcSchemaUID(schema string) []byte {
```

Comment:
If we generate the UID when we're generating code we also don't need this method to re-parse the schema.

Reply:
👍

Reply:
func removed